### PR TITLE
Bugfix: enables the use of builtInPages in the navigation

### LINKED ISSUE #752: Cannot use built-in pages in navigation
**Describe the bug**
Built-in pages used as the internal link reference are filtered out of the navigation in [getTopLevelNavItems](https://github.com/NWACus/web/blob/main/src/components/Header/utils.ts#L242). 

**To Reproduce**

1. Add a built-in page as an internal link's "page or post" in the navigation
2. Visit that tenant's home page (or any page)
3. Observe that the link with the built-in page as the internal link's "page or post" does not appear in the nav

**Expected behavior**
The nav item referencing the built-in page should appear in the nav. It should use a url that is _not_ nested under whatever nav item it's nested under. 

**Screenshots**
https://www.loom.com/share/f157303aeeb34c15a854ad0bf53ac73c

**Environment details**

- Desktop
- Chrome

### PR DESCRIPTION
**Pull request by busbyk**
## Description

`builtInPages` weren't being handled correctly in the navigation. They were being filtered out. This allows them in the nav. 

## Related Issues

Fixes #752

## Key Changes

- Enables the use of built-in pages in the navigation
- Uses a built-in page for the weather stations link in the seed script

## Screenshots / Demo

https://www.loom.com/share/134e95648cae46cf8c2dc75fc04fc5e4

## Changes needed after merge

- [ ] Update all centers' weather stations nav link to use their built-in page
- [ ] Communicate that the local accidents can be a built-in page for Sierra / add to their nav. Won't be a significant difference for them but might as well change it.

### CODE REVIEW

File: src/components/Header/utils.ts [hunk 1]
```diff
@@ -204,13 +207,24 @@ export function convertToNavLink(
...
-    let url =
-      link.reference.relationTo === 'pages'
-        ? `/${reference.value.slug}`
-        : `/blog/${reference.value.slug}`
+    let url: string
+    if (link.reference.relationTo === 'builtInPages' && 'url' in reference.value) {
+      url = normalizePath(reference.value.url, { ensureLeadingSlash: true })
+    } else if (link.reference.relationTo === 'posts' && 'slug' in reference.value) {
+      url = `/blog/${reference.value.slug}`
+    } else if ('slug' in reference.value) {
```
Comment (rchlfryn) [inline src/components/Header/utils.ts]: do we want to add `link.reference.relationTo === 'pages' &&` here as well? That was our initial check and it is not here anymore.

This might also cover external URLs so I could see the case for keeping it out

File: src/endpoints/seed/navigation.ts [hunk 2]
```diff
@@ -45,6 +46,34 @@ export const navigationSeed = (
     }
   }
 
+  const builtInPageLink = ({
```
Comment (rchlfryn) [inline src/endpoints/seed/navigation.ts:49]: 👏🏼 this was needed

File: src/endpoints/seed/navigation.ts [hunk 3]
```diff
@@ -45,6 +46,34 @@ export const navigationSeed = (
...
+    url,
+    label,
+  }: {
+    url: string
+    label?: string
+  }): NonNullable<NonNullable<Navigation['donate']>['link']> => {
```
Comment (rchlfryn) [inline src/endpoints/seed/navigation.ts]: I think this is the wrong type? `Navigation['donate']>['link']`

File: src/endpoints/seed/navigation.ts [hunk 4]
```diff
@@ -45,6 +46,34 @@ export const navigationSeed = (
...
+
+    const builtInPage = builtInPages[tenant.slug][url]
+
+    return {
+      type: 'internal',
+      reference: {
+        value: builtInPage.id,
+        relationTo: 'builtInPages',
+      },
+      label: label || builtInPage.title,
```
Comment (rchlfryn) [inline src/endpoints/seed/navigation.ts:73]: Since label is optional, do we want to add a default assignment here? It might be nice to test the edge case.

File: src/components/Header/utils.ts [hunk 5]
```diff
@@ -176,7 +177,9 @@ export function convertToNavLink(
   if (
     link.type === 'internal' &&
     link.reference &&
-    (link.reference.relationTo === 'pages' || link.reference.relationTo === 'posts')
+    (link.reference.relationTo === 'pages' ||
+      link.reference.relationTo === 'posts' ||
+      link.reference.relationTo === 'builtInPages')
```
Comment (rchlfryn) [inline src/components/Header/utils.ts]: It is interesting we allow and combine all of these type options only to separate them later in 210. Should we just separate them form here? I am not sure what is cleaner.

**Review by rchlfryn — CHANGES_REQUESTED** [review]
A few thoughts.. good call on this PR

File: src/components/Header/utils.ts (hunk 1 above)
Comment (busbyk) [reply src/components/Header/utils.ts]: Good call. I like being more specific here. I'll add that back.

File: src/endpoints/seed/navigation.ts (hunk 3 above)
Comment (busbyk) [reply src/endpoints/seed/navigation.ts]: Ope, yea double nested NonNullables. We need this one NonNullable though because Navigation['donate'] is possibly undefined.

File: src/endpoints/seed/navigation.ts (hunk 4 above)
Comment (busbyk) [reply src/endpoints/seed/navigation.ts:73]: I.e. if builtInPage.title is null?

File: src/components/Header/utils.ts (hunk 5 above)
Comment (busbyk) [reply src/components/Header/utils.ts]: What do you think about this refactor? [1f91195](https://github.com/NWACus/web/pull/781/commits/1f91195c357649c809685caed3d12eebc721bc99)

Tried to make this a little bit easier to read / cleaner.

File: src/endpoints/seed/navigation.ts (hunk 4 above)
Comment (rchlfryn) [reply src/endpoints/seed/navigation.ts:73]: I was just saying it is optional on line 54 so it seems weird to have a default here but I get why you did it

File: src/components/Header/utils.ts (hunk 5 above)
Comment (rchlfryn) [reply src/components/Header/utils.ts]: Awesome!

**Review by rchlfryn — APPROVED** [review]
